crates/astria-sequencer/src/checked_actions/bridge/bridge_transfer.rs

```diff
@@ -0,0 +1,624 @@
+        TransactionId,
+        ADDRESS_LEN,
+    },
+    protocol::transaction::v1::action::{
+        BridgeLock,
+        BridgeTransfer,
+        BridgeUnlock,
+    },
+};
+use astria_eyre::eyre::{
+    Result,
+    WrapErr as _,
+};
+use cnidarium::{
+    StateRead,
+    StateWrite,
+};
+use tracing::{
+    instrument,
+    Level,
+};
+
+use super::{
+    super::AssetTransfer,
+    bridge_lock::CheckedBridgeLockImpl,
+    bridge_unlock::CheckedBridgeUnlockImpl,
+};
+use crate::accounts::StateWriteExt as _;
+
+#[derive(Debug)]
+pub(crate) struct CheckedBridgeTransfer {
+    action: BridgeTransfer,
+    checked_bridge_unlock: CheckedBridgeUnlockImpl<false>,
+    checked_bridge_lock: CheckedBridgeLockImpl<false>,
+}
+
+impl CheckedBridgeTransfer {
+    #[instrument(skip_all, err(level = Level::DEBUG))]
+    pub(in crate::checked_actions) async fn new<S: StateRead>(
+        action: BridgeTransfer,
+        tx_signer: [u8; ADDRESS_LEN],
+        tx_id: TransactionId,
+        position_in_tx: u64,
+        state: S,
+    ) -> Result<Self> {
+        let BridgeTransfer {
+            to,
+            amount,
+            fee_asset,
+            destination_chain_address,
+            bridge_address,
+            rollup_block_number,
+            rollup_withdrawal_event_id,
+        } = action.clone();
+
+        let bridge_unlock = BridgeUnlock {
+            to,
+            amount,
+            memo: String::new(),
+            rollup_withdrawal_event_id,
+            rollup_block_number,
+            fee_asset: fee_asset.clone(),
+            bridge_address,
+        };
+        let checked_bridge_unlock =
+            CheckedBridgeUnlockImpl::<false>::new(bridge_unlock, tx_signer, &state)
+                .await
+                .wrap_err("failed to construct checked bridge unlock for bridge transfer")?;
+
+        let bridge_lock = BridgeLock {
+            to,
+            amount,
+            asset: checked_bridge_unlock.bridge_account_ibc_asset().into(),
+            fee_asset,
+            destination_chain_address,
+        };
+        let checked_bridge_lock = CheckedBridgeLockImpl::<false>::new(
+            bridge_lock,
+            tx_signer,
+            tx_id,
+            position_in_tx,
+            &state,
+        )
+        .await
+        .wrap_err("failed to construct checked bridge lock for bridge transfer")?;
+
+        let checked_action = Self {
+            action,
+            checked_bridge_unlock,
+            checked_bridge_lock,
+        };
+        checked_action.run_mutable_checks(state).await?;
+
+        Ok(checked_action)
+    }
+
+    #[instrument(skip_all, err(level = Level::DEBUG))]
+    pub(in crate::checked_actions) async fn run_mutable_checks<S: StateRead>(
+        &self,
+        state: S,
+    ) -> Result<()> {
+        self.checked_bridge_unlock
+            .run_mutable_checks(&state)
+            .await
+            .wrap_err("mutable checks for bridge unlock failed for bridge transfer")?;
```

Comment:
same here, bridge transfers are intended to be from bridge account to bridge account, so also remove this and ensure the receive is a bridge account